sklearn/cross_validation.py

```diff
@@ -1189,8 +1189,10 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params):
     # Adjust length of sample weights
     n_samples = _num_samples(X)
     fit_params = fit_params if fit_params is not None else {}
-    fit_params = dict([(k, np.asarray(v)[train]
-                       if hasattr(v, '__len__') and len(v) == n_samples else v)
+    fit_params = dict([(k, (v.tocsr()[train] if sp.issparse(v)
```

Comment:
@akshayah3 I think this comment was not addressed. I also think we should use `safe_indexing` utility would do the job here (although the call to `toscsr()` is probably still required). 

Speaking of which, the `safe_indexing` utility could raise a more meaning exception when calling it on a sparse array that does not support row-based indexing such as COO or CSC. (CSR, LIL and DOK all support row based indexing apparently).